model/task/task.go

```diff
@@ -1220,27 +1229,59 @@ func GenerateNotRun() ([]Task, error) {
 	}))
 }
 
-// SetGeneratedJSON sets JSON data to generate tasks from.
-func (t *Task) SetGeneratedJSON(json []json.RawMessage) error {
-	if len(t.GeneratedJSONAsString) > 0 {
+// SetGeneratedJSON sets JSON data to generate tasks from. If the generated JSON
+// files have already been stored, this is a no-op.
+func (t *Task) SetGeneratedJSON(files GeneratedJSONFiles) error {
+	if len(t.GeneratedJSONAsString) > 0 || t.GeneratedJSONStorageMethod != "" {
 		return nil
 	}
-	s := []string{}
-	for _, j := range json {
-		s = append(s, string(j))
+
+	if err := UpdateOne(
+		bson.M{
+			IdKey:                         t.Id,
+			GeneratedJSONAsStringKey:      bson.M{"$exists": false},
+			GeneratedJSONStorageMethodKey: nil,
+		},
+		bson.M{
+			"$set": bson.M{
+				GeneratedJSONAsStringKey:      files,
+				GeneratedJSONStorageMethodKey: evergreen.ProjectStorageMethodDB,
+			},
+		},
+	); err != nil {
+		return err
 	}
-	t.GeneratedJSONAsString = s
-	return UpdateOne(
+
+	t.GeneratedJSONAsString = files
+	t.GeneratedJSONStorageMethod = evergreen.ProjectStorageMethodDB
+
+	return nil
+}
+
+// SetGeneratedJSONStorageMethod sets the task's generated JSON file storage
+// method. If it's already been set, this is a no-op.
+func (t *Task) SetGeneratedJSONStorageMethod(method evergreen.ParserProjectStorageMethod) error {
+	if t.GeneratedJSONStorageMethod != "" {
```

Comment:
Inserting the generated JSON files is an all-or-nothing operation (since both this + `SetGeneratedJSON` atomically set the field) so I don't think we need to narrow it down to the method passed in. In any case, it doesn't actually matter where it's stored, only it is stored in one place (which the presence of this field will indicate).